model/src/main/kotlin/utils/DependencyGraphBuilder.kt

```diff
@@ -331,9 +331,19 @@ class DependencyGraphBuilder<D>(
         val dependencies = dependencyHandler.dependenciesFor(dependency)
         if (ref.dependencies.size != dependencies.size) return false
 
-        val dependencies1 = ref.dependencies.map { dependencyIds[it.pkg] }
+        val dependencies1 = ref.dependencies.mapTo(mutableSetOf()) { dependencyIds[it.pkg] }
         val dependencies2 = dependencies.associateBy { dependencyHandler.identifierFor(it) }
-        if (!dependencies2.keys.containsAll(dependencies1)) return false
+
+        if (dependencies1 == dependencies2.keys) {
```

Comment:
I believe that no "GradleDependencyHandler" specific code must be put here.
Instead we should provide means to  inject / override the equality comparison function.

Reply:
I introduced `requiresDeepDependencyTreeComparison()` to the interface. Feel free to rename.